libs/SmartSync/SmartSync/Classes/Util/SFSyncServerTarget.m

```diff
@@ -0,0 +1,176 @@
+/*
+ Copyright (c) 2015, salesforce.com, inc. All rights reserved.
+ 
+ Redistribution and use of this software in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list of conditions
+ and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "SFSyncServerTarget.h"
+#import "SFSmartSyncConstants.h"
+#import "SFSmartSyncObjectUtils.h"
+#import "SFSmartSyncSoqlBuilder.h"
+#import "SFSmartSyncNetworkUtils.h"
+#import <SalesforceRestAPI/SFRestAPI+Blocks.h>
+#import <SalesforceRestAPI/SFRestRequest.h>
+#import <SalesforceSDKCore/SFJsonUtils.h>
+
+// target types
+static NSString * const kSFSyncServerTargetTypeRestStandard = @"rest";
+static NSString * const kSFSyncServerTargetTypeCustom = @"custom";
+
+@implementation SFSyncServerTarget
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        self.targetType = SFSyncServerTargetTypeRestStandard;
+    }
+    return self;
+}
+
+#pragma mark - Serialization and factory methods
+
++ (instancetype)newFromDict:(NSDictionary*)dict {
+    NSString *implClassName;
+    NSString *targetTypeString = (dict[kSFSyncTargetTypeKey] == nil ? kSFSyncServerTargetTypeRestStandard : dict[kSFSyncTargetTypeKey]);
+    switch ([self targetTypeFromString:targetTypeString]) {
+        case SFSyncServerTargetTypeCustom:
+            implClassName = dict[kSFSyncTargetiOSImplKey];
+            return [NSClassFromString(implClassName) newFromDict:dict];
```

Comment:
I basically copied this implementation from `SFSyncTarget`, but it seems like an anti-pattern.  We're breaking de facto inheritance here.  Shouldn't this be calling into a common initializer, like `initWithDict`?